--- a/config/routes	2015-04-18 17:56:17.664145973 +0200
+++ b/config/routes	2015-04-18 17:57:42.732147137 +0200
@@ -1,7 +1,10 @@
 /static StaticR Static appStatic
-/auth   AuthR   Auth   getAuth
 
 /favicon.ico FaviconR GET
 /robots.txt RobotsR GET
 
 / HomeR GET POST
+
+/admin/auth   AuthR                Auth         getAuth
+/admin/core   CoreAdminR           CoreAdmin    getLambdaCms
+/admin        AdminHomeRedirectR   GET
