--- a/project_name.cabal	2015-04-18 17:56:16.688145960 +0200
+++ b/project_name.cabal	2015-04-18 18:30:32.280020035 +0200
@@ -22,6 +22,7 @@
                      Settings.StaticFiles
                      Handler.Common
                      Handler.Home
+                     Roles
 
     if flag(dev) || flag(library-only)
         cpp-options:   -DDEVELOPMENT
@@ -81,6 +82,8 @@
                  , containers
                  , vector
                  , time
+                 , lambdacms-core                >= 0.1.0      && < 0.2
+                 , wai                           >= 3.0.2      && < 3.1
 
 executable         mysite
     if flag(library-only)
