--- a/Settings.hs	2015-04-18 17:56:17.664145973 +0200
+++ b/Settings.hs	2015-04-18 20:05:12.496097754 +0200
@@ -52,6 +52,9 @@
     -- ^ Copyright text to appear in the footer of the page
     , appAnalytics              :: Maybe Text
     -- ^ Google Analytics code
+
+    , appAdmin                  :: Text
+    -- ^ LambdaCms default admin email address
     }
 
 instance FromJSON AppSettings where
@@ -78,6 +81,8 @@
         appCopyright              <- o .: "copyright"
         appAnalytics              <- o .:? "analytics"
 
+        appAdmin                  <- o .: "admin"
+
         return AppSettings {..}
 
 -- | Settings for 'widgetFile', such as which template languages to support and
