--- a/config/routes	2015-07-22 22:52:20.275915255 +0200
+++ b/config/routes	2015-07-23 00:41:55.817156642 +0200
@@ -1,7 +1,10 @@
 /static StaticR Static appStatic
-/auth   AuthR   Auth   getAuth
 
 /favicon.ico FaviconR GET
 /robots.txt RobotsR GET
 
-/ HomeR GET POST
+/ HomeR GET
+
+/admin/auth   AuthR                Auth         getAuth
+/admin        AdminHomeRedirectR   GET
+/admin/core   CoreAdminR           CoreAdmin    getLambdaCms
