--- a/config/routes	2015-07-22 14:06:58.168019891 +0200
+++ b/config/routes	2015-07-22 14:06:59.744020066 +0200
@@ -1,7 +1,10 @@
 /static StaticR Static appStatic
-/auth   AuthR   Auth   getAuth
 
 /favicon.ico FaviconR GET
 /robots.txt RobotsR GET
 
 / HomeR GET POST
+
+/admin/auth   AuthR                Auth         getAuth
+/admin/core   CoreAdminR           CoreAdmin    getLambdaCms
+/admin        AdminHomeRedirectR   GET
