--- a/project_name.cabal	2015-07-19 17:26:17.398712467 +0200
+++ b/project_name.cabal	2015-07-19 15:54:09.416756403 +0200
@@ -22,7 +22,6 @@
                      Settings.StaticFiles
                      Handler.Common
                      Handler.Home
-                     Roles
 
     if flag(dev) || flag(library-only)
         cpp-options:   -DDEVELOPMENT
@@ -82,8 +81,6 @@
                  , containers
                  , vector
                  , time
-                 , wai                           >= 3.0.2      && < 3.1
-                 , lambdacms-core                >= 0.3.0.0    && < 0.4
 
 executable         mysite
     if flag(library-only)
