--- a/test/TestImport.hs	2015-07-22 22:52:20.271915262 +0200
+++ b/test/TestImport.hs	2015-07-22 22:52:23.023910152 +0200
@@ -10,7 +10,6 @@
 import Foundation            as X
 import Model                 as X
 import Test.Hspec            as X
-import Text.Shakespeare.Text (st)
 import Yesod.Default.Config2 (ignoreEnv, loadAppSettings)
 import Yesod.Test            as X
 
@@ -49,8 +48,8 @@
 
     -- Aside: SQLite by default *does not enable foreign key checks*
     -- (disabling foreign keys is only necessary for those who specifically enable them).
-    let settings = appSettings app   
-    sqliteConn <- rawConnection (sqlDatabase $ appDatabaseConf settings)    
+    let settings = appSettings app
+    sqliteConn <- rawConnection (sqlDatabase $ appDatabaseConf settings)
     disableForeignKeys sqliteConn
 
     let logFunc = messageLoggerSource app (appLogger app)
